test/System.Net.Http.Formatting.Test/HttpContentMessageExtensionsTests.cs

```diff
@@ -482,16 +482,50 @@ public Task ReadAsHttpResponseMessageAsync_LargeHeaderSize()
         }
 
         [Fact]
-        public Task ReadAsHttpRequestMessageAsync_LargeHeaderSize()
+        public async Task ReadAsHttpRequestMessageAsync_LargeHeaderSize()
         {
+            string cookieValue = string.Format("{0}={1}", new String('a', 16 * 1024), new String('b', 16 * 1024));
             string[] request = new[] {
                 @"GET / HTTP/1.1",
                 @"Host: msdn.microsoft.com",
-                String.Format("Cookie: {0}={1}", new String('a', 16 * 1024), new String('b', 16 * 1024))
+                string.Format("Cookie: {0}", cookieValue),
+            };
+
+            HttpContent content = CreateContent(true, request, "sample body");
+            var httpRequestMessage = await content.ReadAsHttpRequestMessageAsync(Uri.UriSchemeHttp, 64 * 1024, 64 * 1024);
+
+            Assert.Equal(HttpMethod.Get, httpRequestMessage.Method);
+            Assert.Equal("/", httpRequestMessage.RequestUri.PathAndQuery);
+            Assert.Equal("msdn.microsoft.com", httpRequestMessage.Headers.Host);
+            IEnumerable<string> actualCookieValue;
```

Comment:
bet you're missing `out var` 😀

Reply:
I tried that and it does not work in C# 5. This is like getting -ve 💯 points.